src/kafunk/ProduceBuffer.fs

```diff
@@ -0,0 +1,49 @@
+﻿namespace Kafunk
+
+open FSharp.Control
+open Kafunk
+
+type BufferingProducer  = private {
+
+  ///The producer
+  producer : Producer
+
+  ///The buffer
+  buffer : Buffer<ProducerMessage>
+  
+  consume : ProducerMessage -> bool }
+
+and BufferType = 
+    | Blocking
+    | Discarding
+
+[<Compile(Module)>]
+module BufferingProducer = 
+
+  let private Log = Log.create "Kafunk.BufferingProducer"
+  
+  ///Create a producer with buffer
+  let createBufferingProducer (producer:Producer) (buffertype:BufferType) (capacity:int) (batchSize:int) (batchTimeMs:int) (timeIntervalMs:int) (errorHandle:(ProducerMessage seq -> unit) option)= 
+
+    let consume (producer: Producer) (batch: ProducerMessage seq): Async<unit> = async {
+      try
+        do! Producer.produceBatched producer batch |> Async.Ignore
+        return ()
+      with ex ->
+        Log.error "buffering_producer_process_error|error=\"%O\" topic=%s" ex producer.config.topic
```

Comment:
This should log, but also raise an Event along with the messages that failed to process. This way, they can potentially be logged.